crates/runtime/src/podswatcher.rs

```diff
@@ -91,6 +90,19 @@ impl PodsWatcher {
     }
 }
 
+macro_rules! enable_watch_for_component {
```

Comment:
Also considered the following approach but chaining is not possible as ComponentOrReference items are of different types.

```
let all_components = chain!(
    spicepod.datasets.iter(),
    spicepod.models.iter(),
    spicepod.catalogs.iter(),
    spicepod.views.iter(),
    spicepod.embeddings.iter(),
    spicepod.evals.iter(),
    spicepod.tools.iter(),
);

for component in all_components {
    match component {
        ComponentOrReference::Reference(reference) => {
            let ref_path = root_dir.join(&reference.r#ref);
            dirs.push(root_dir.join(ref_path));
        }
        ComponentOrReference::Component(_) => { /* ignore component */ }
    }
}
```